benchmark/torch/mappo/mappo_model.py

```diff
@@ -0,0 +1,119 @@
+#   Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# modified from https://github.com/marlbenchmark/on-policy
+
+import parl
+import torch
+import torch.nn as nn
+import numpy as np
+import torch.nn.functional as F
+from mappo_buffer import get_shape_from_obs_space
+
+
+def check(input):
+    output = torch.from_numpy(input) if type(input) == np.ndarray else input
+    return output
+
+
+class MAPPOModel(parl.Model):
+    def __init__(self,
+                 obs_space,
+                 cent_obs_space,
+                 act_space,
+                 device=torch.device("cpu")):
+        super(MAPPOModel, self).__init__()
+        self.device = device
+        self.obs_space = obs_space
+        self.share_obs_space = cent_obs_space
+        self.act_space = act_space
+
+        self.actor = Actor(self.obs_space, self.act_space, self.device)
+        self.critic = Critic(self.share_obs_space, self.device)
+
+    def policy(self, obs, available_actions=None, deterministic=False):
+        actions = self.actor(obs, available_actions, deterministic)
+        return actions
+
+    def value(self, cent_obs):
+        values = self.critic(cent_obs)
+        return values
+
+
+class Actor(parl.Model):
+    def __init__(self, obs_space, action_space, device=torch.device("cpu")):
+        super(Actor, self).__init__()
+        self.tpdv = dict(dtype=torch.float32, device=device)
+        self.multi_discrete = False
+        obs_shape = get_shape_from_obs_space(obs_space)
```

Comment:
Please use `get_shape_from_obs_space` in train.py, just simply pass the integer `obs_dim` instead of the class `obs_space` to Model class